CONTRIBUTING.md

```diff
@@ -0,0 +1,177 @@
+  * If the bug is caused by an external file (ex: when importing subscriptions), try to include
+    that too (Get in touch with us if you want to share these files privately).
+  * If you're hosting your own instance, include relevant config file(s).\
+    **Make sure to redact secrets like your DB password or HMAC key first!!**
+
+
+**Note: Security-related issues should be reported by e-mail at
+[security@invidious.io](mailto:security@invidious.io).**
+
+
+### Enhancement
+
+If you feel that some existing feature can be improved, you should open an "enhancement" issue.
+
+In your issue, describe what the Invidious currently does, what you don't like about it, and propose
+ways to improve that feature. If you can provide screenshots or drawings to support your explanation
+that's even better!
+
+Please be aware that Invidious is heavily aimed towards simplicity and being usable without
+Javascript. It means that we have to deal with compromises all over the place, and some features
+might not be as good as JavaScript-rich alternatives (like Freetube), that allows much more
+flexibility.
+
+
+### Feature resquests
+
+If you think that Invidious lacks some feature or another, you should open a "feature request".
+
+Do note that Invidious is and will remain a Youtube-only front-end. Requests to add support for
+other services (e.g: Bandcamp, [Odyssee](https://github.com/iv-org/invidious/issues/3022)) will be
+systematically rejected.
+
+
+
+## Translations
+
+Invidious is translated in many languages using Weblate:
+https://hosted.weblate.org/projects/invidious/.
+
+We recommend creating an account or connecting with one of the other authentication providers that
+Weblate supports (Github, GitLab, Google, etc..) for a better experience.
+
+We also accept translation updates using Pull requests, but please be aware that it represents more
+work for us to merge those.
+
+
+
+## Hosting an instance
+
+Another way to contribute to invidious is to host a [public instance]
+(https://instances.invidious.io/).
+
+To do so, you need a server (either a VPS or dedicated) with
+[enough ressources](https://docs.invidious.io/installation/#hardware-requirements) to handle the
+load. You'll also need a domain name with a valid TLS certificate (e.g provided by Let's Encrypt).
+
+Then, if your instance follows the
+[rules listed here](https://docs.invidious.io/instances/#rules-to-have-your-instance-in-this-list),
+you can request your instance to be added to the list by
+[creating an issue on the documentation repository](https://github.com/iv-org/documentation/issues).
+
+Once you've filled the form with your instance's informations, your instance will be added to our
+uptime monitor. From there, a probatory period of 30 days will start, to make sure that you can
+keep your instance online and up to date. Finally, your instance will be added to the list. We'll
+ask you to join our Matrix room, so that we can inform you of important updates and exchange with
```

Comment:
```suggestion
ask you to join our Matrix room, so that we can inform you of important updates, warn you in advance of security issues and exchange with
```